tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Type Driven Development in Python
+published: 2019-08-01
+tags: Python, typing, types, tdd
+language: python
+author-name: Carlos Villavicencio
+twitter-profile: po5i
+github-profile: po5i
+description: Learn how to improve the software quality and readability by using type driven development in Python.
+---
+
+In a [previous article][types-vs-tests], we discussed the benefits of working with Type Driven Development and the importance of static type checking in compilation time for strongly typed languages such as Java or Haskell. It doesn’t matter if their paradigm is object-oriented or functional,static types add a layer of quality, allows us to reduce the amount of tests we need to write, and reduces our dependency on tests.
```

Comment:
@po5i I think is good that we reference a previous article but it might be an indirection for readers if we start pointing to a separate article first. Perhaps we could reorder things a little bit since on the second sentence you emphasize the importance of the safety provided by the static analysis which I think is a good start point to engage readers since the beginning. What do you think?

Additionally a couple of minor suggestions related to this paragraph:

```diff
- doesn't matter if their paradigm
+ doesn't matter if the paradigm
```

```diff
- static types add a layer of quality, allows us to reduce the amount
+ type-checking adds a layer of safety, reducing the amount
```

Reply:
@po5i `and reduces our dependency on tests` I think this is a good hypothesis, maybe we should take that into account during the conclusions, for instance from the reader standpoint I would be interested in learning more about what kind of specific test I could get rid off by adding types to my code.